boot/boot_serial/src/boot_serial.c

```diff
@@ -339,8 +370,14 @@ boot_serial_output(void)
     bs_hdr->nh_len = htons(len);
     bs_hdr->nh_group = htons(bs_hdr->nh_group);
 
+#ifdef __ZEPHYR__
+    crc =  crc16((u8_t *)bs_hdr, sizeof(*bs_hdr), CRC_CITT_POLYMINAL, CRC16_INITIAL_CRC,
```

Comment:
FYI:https://github.com/zephyrproject-rtos/zephyr/pull/5098/files

Reply:
Actually, you should use the one in the header and then you'll get the faster version once that PR is merged:

https://github.com/zephyrproject-rtos/zephyr/blob/master/include/crc16.h#L47

Reply:
The problem here is that crc is calculated form two parts, so 1st crc is initial value for 2nd calculation and 1st calculation is without padding. Current and pointed implementation of ciit crc make such calculation impossible.
But maybe ther is a mathematical feature of CRC that you know, but I don't know.